### ISSUE TITLE
Config to generate server interface only

### ISSUE THREAD

**Issue by dnguyenzd** [deepmap/oapi-codegen#1630]
hi, I'm wondering whether it's possible to add a generate config to generate `ServerInterface` only? something like:

```yaml
generate:
  server-interface: true  # imply `models: true`
```

context: we use chi as the router and use chi sub router to organise different groups of routes with different middlewares. From what I see in [this issue](https://github.com/deepmap/oapi-codegen/issues/1279), this scenario is not well supported by `oapi-codegen` so we choose to not use the generated `ServerInterfaceWrapper` and register routes by ourselves intead. But this causes a lot of unused generated code

ideally we just need the models and `ServerInterface` interface (so that we can do type checking and make sure the handlers satisfy the interface)

**Comment by jamietanna** [comment]
Interesting, so you're looking to just have the server interface but not the code to register that handler with a Chi router?

**Comment by dnguyenzd** [comment]
yes, that's right. We'll do the registration by ourselves to take advantages of other features of Chi
